applications/crossbar/src/modules/cb_push_notification_subscriptions.erl

```diff
@@ -0,0 +1,215 @@
+%% and load necessary information.
+%% /push_notification_subscriptions mights load a list of push_notificatio_subscriptionn objects
+%% /push_notification_subscriptions/123 might load the push_notificatio_subscriptionn object 123
+%% Generally, use crossbar_doc to manipulate the cb_context{} record
+%% @end
+%%------------------------------------------------------------------------------
+-spec validate(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+validate(Context, App, MobileDeviceId) ->
+    validate_push_notification_subscription(Context, App, MobileDeviceId, cb_context:req_verb(Context)).
+
+-spec validate_push_notification_subscription(cb_context:context(), path_token(), path_token(), http_method()) -> cb_context:context().
+validate_push_notification_subscription(Context, App, MobileDeviceId, ?HTTP_PUT) ->
+    lager:debug("validating push notification subscription for PUT"),
+    create(Context, App, MobileDeviceId);
+validate_push_notification_subscription(Context, App, MobileDeviceId, ?HTTP_POST) ->
+    lager:debug("validating push notification subscription for POST"),
+    update(App, MobileDeviceId, Context);
+validate_push_notification_subscription(Context, App, MobileDeviceId, ?HTTP_DELETE) ->
+    lager:debug("validating push notification subscription for DELETE"),
+    read_docs_for_app_and_mobile_device_id(Context, App, MobileDeviceId);
+validate_push_notification_subscription(Context, App, MobileDeviceId, ?HTTP_GET) ->
+    lager:debug("validating push notification subscription for GET"),
+    read_docs_for_app_and_mobile_device_id(Context, App, MobileDeviceId).
+
+%%------------------------------------------------------------------------------
+%% @doc If the HTTP verb is PUT, execute the actual action, usually a db save.
+%% @end
+%%------------------------------------------------------------------------------
+-spec put(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+put(Context, _App, _MobileDeviceId) ->
+    %% Set the app on the doc
+    crossbar_doc:save(Context).
+
+%%------------------------------------------------------------------------------
+%% @doc If the HTTP verb is POST, execute the actual action, usually a db save
+%% (after a merge perhaps).
+%% @end
+%%------------------------------------------------------------------------------
+-spec post(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+post(Context, _, _) ->
+    crossbar_doc:save(Context).
+
+%%------------------------------------------------------------------------------
+%% @doc If the HTTP verb is DELETE, execute the actual action, usually a db delete
+%% @end
+%%------------------------------------------------------------------------------
+-spec delete(cb_context:context(), path_token(), path_token()) -> cb_context:context().
+delete(Context, _, _) ->
+    Id = kz_json:get_value(<<"id">>, cb_context:doc(Context)),
```

Comment:
Why was `crossbar_doc:delete(Context).` not used?

Reply:
I first tried using `crossbar_doc:delete` but that didn't work, the request would return successfully and say the document has been deleted but really it was still in the database without `pvt_deleted` being set and could be retrieved with subsequent gets.